### PR TITLE
Replacing Guava with java collections

### PR DESCRIPTION
**Pull request by DavidRG13**
Part of #2212

Continue the migration

### CODE REVIEW

File: webapp-mgmt/cas-management-webapp-support/src/test/java/org/apereo/cas/services/web/RegisteredServiceSimpleFormControllerTests.java [hunk 1]
```diff
@@ -135,9 +134,7 @@ public void verifyAddRegexRegisteredService() throws Exception {
         svc.setEvaluationOrder(1000);
 
        final RegisteredServiceEditBean.ServiceData data = registeredServiceFactory.createServiceData(svc);
-       this.controller.saveService(new MockHttpServletRequest(),
-               new MockHttpServletResponse(),
-               data, mock(BindingResult.class));
+       this.controller.saveService(new MockHttpServletRequest(), new MockHttpServletResponse(), data, mock(BindingResult.class));
```
Comment (mmoayyed) [inline webapp-mgmt/cas-management-webapp-support/src/test/java/org/apereo/cas/services/web/RegisteredServiceSimpleFormControllerTests.java]: Is this a legit complaint??

Comment (DavidRG13) [inline webapp-mgmt/cas-management-webapp-support/src/test/java/org/apereo/cas/services/web/RegisteredServiceSimpleFormControllerTests.java]: Probably... just added a commit formatting it ;)

File: core/cas-server-core-services/src/main/java/org/apereo/cas/services/DefaultRegisteredServiceAccessStrategy.java [hunk 2]
```diff
@@ -230,91 +230,63 @@ public boolean doPrincipalAttributesAllowServiceAccess(final String principal, f
     private boolean doRequiredAttributesAllowPrincipalAccess(final Map<String, Object> principalAttributes) {
         LOGGER.debug("These required attributes [{}] are examined against [{}] before service can proceed.",
                 this.requiredAttributes, principalAttributes);
-        final Sets.SetView<String> difference = Sets.intersection(this.requiredAttributes.keySet(), principalAttributes.keySet());
-        final Set<String> copy = difference.immutableCopy();
+        final Set<String> difference = this.requiredAttributes.keySet().stream()
```
Comment (mmoayyed) [inline core/cas-server-core-services/src/main/java/org/apereo/cas/services/DefaultRegisteredServiceAccessStrategy.java]: Is this something that can be done with `retailAll` instead?

File: core/cas-server-core-services/src/main/java/org/apereo/cas/services/DefaultRegisteredServiceAccessStrategy.java [hunk 3]
```diff
@@ -412,4 +384,13 @@ public String toString() {
                 .append("rejectedAttributes", this.rejectedAttributes)
                 .toString();
     }
+
+    private static Set<String> objectToSetOfStrings(final Object objVal) {
```
Comment (mmoayyed) [inline core/cas-server-core-services/src/main/java/org/apereo/cas/services/DefaultRegisteredServiceAccessStrategy.java]: Move this to CollectionUtils. In fact, there may be something similar there already?

File: core/cas-server-core-services/src/main/java/org/apereo/cas/services/DefaultRegisteredServiceAccessStrategy.java [hunk 4]
```diff
@@ -230,91 +230,63 @@ public boolean doPrincipalAttributesAllowServiceAccess(final String principal, f
...
+                .filter(a -> principalAttributes.keySet().contains(a))
+                .collect(Collectors.toSet());
 
         if (this.rejectedAttributes.isEmpty()) {
             LOGGER.debug("No rejected attributes are defined");
             return false;
         }
         
-        if (this.requireAllAttributes && rejectedCopy.size() < this.rejectedAttributes.size()) {
+        if (this.requireAllAttributes && rejectedDifference.size() < this.rejectedAttributes.size()) {
```
Comment (mmoayyed) [inline core/cas-server-core-services/src/main/java/org/apereo/cas/services/DefaultRegisteredServiceAccessStrategy.java]: It seems like the two blocks for rejectedDifference and difference collections are almost identical. Can we refactor and reuse?

File: core/cas-server-core-services/src/main/java/org/apereo/cas/services/DefaultRegisteredServiceAccessStrategy.java [hunk 5]
```diff
@@ -412,4 +384,13 @@ public String toString() {
                 .append("rejectedAttributes", this.rejectedAttributes)
                 .toString();
     }
+
+    private static Set<String> objectToSetOfStrings(final Object objVal) {
+        if (objVal instanceof Collection) {
+            final Collection collection = (Collection) objVal;
+            return new HashSet<>(collection);
+        } else {
+            return Collections.singleton(objVal.toString());
```
Comment (mmoayyed) [inline core/cas-server-core-services/src/main/java/org/apereo/cas/services/DefaultRegisteredServiceAccessStrategy.java]: Also not a big deal, but the else clause is rather redundant.  See https://apereo.github.io/cas/developer/Code-Conventions.html#needless-else-clauses

Comment (mmoayyed) [inline core/cas-server-core-services/src/main/java/org/apereo/cas/services/DefaultRegisteredServiceAccessStrategy.java]: Probably also something we could build into checkstyle rules to keep everything in one format.

File: core/cas-server-core-services/src/main/java/org/apereo/cas/services/DefaultRegisteredServiceAccessStrategy.java (hunk 4 above)
Comment (mmoayyed) [inline core/cas-server-core-services/src/main/java/org/apereo/cas/services/DefaultRegisteredServiceAccessStrategy.java]: Also, there probably is plenty but please verify there are unit tests that thoroughly test the changes in this area.

File: core/cas-server-core-services/src/test/java/org/apereo/cas/authentication/DefaultMultifactorTriggerSelectionStrategyTests.java [hunk 6]
```diff
@@ -168,6 +169,6 @@ private static RegexRegisteredService mockPrincipalService(final String provider
     
     private Principal mockPrincipal(final String attrName, final String... attrValues) {
```
Comment (mmoayyed) [inline core/cas-server-core-services/src/test/java/org/apereo/cas/authentication/DefaultMultifactorTriggerSelectionStrategyTests.java]: Could this be moved to CoreAuthenticationTestUtils? I vaguely remember a method call in there that mocks principal creation?

File: core/cas-server-core-services/src/main/java/org/apereo/cas/services/DefaultRegisteredServiceAccessStrategy.java (hunk 2 above)
Comment (DavidRG13) [inline core/cas-server-core-services/src/main/java/org/apereo/cas/services/DefaultRegisteredServiceAccessStrategy.java]: I can change if you want to, but I, personally, don't like the way you have to use ```ratainAll```:
```
final Set<String> difference = new HashSet<>(this.requiredAttributes.keySet());
        difference.retainAll(principalAttributes.keySet());
```
It looks like 2 separate operations, when it's only 1, for me stream is much clearer

File: core/cas-server-core-services/src/test/java/org/apereo/cas/authentication/DefaultMultifactorTriggerSelectionStrategyTests.java (hunk 6 above)
Comment (DavidRG13) [inline core/cas-server-core-services/src/test/java/org/apereo/cas/authentication/DefaultMultifactorTriggerSelectionStrategyTests.java]: should we do it in a separate PR? just to keep it on topic

Comment (mmoayyed) [inline core/cas-server-core-services/src/test/java/org/apereo/cas/authentication/DefaultMultifactorTriggerSelectionStrategyTests.java]: Sure.

Comment (DavidRG13) [inline core/cas-server-core-services/src/test/java/org/apereo/cas/authentication/DefaultMultifactorTriggerSelectionStrategyTests.java]: #2243 created

File: core/cas-server-core-services/src/main/java/org/apereo/cas/services/DefaultRegisteredServiceAccessStrategy.java (hunk 5 above)
Comment (DavidRG13) [inline core/cas-server-core-services/src/main/java/org/apereo/cas/services/DefaultRegisteredServiceAccessStrategy.java]: not needed else removed

File: core/cas-server-core-services/src/main/java/org/apereo/cas/services/DefaultRegisteredServiceAccessStrategy.java (hunk 3 above)
Comment (DavidRG13) [inline core/cas-server-core-services/src/main/java/org/apereo/cas/services/DefaultRegisteredServiceAccessStrategy.java]: moved

File: core/cas-server-core-services/src/main/java/org/apereo/cas/services/DefaultRegisteredServiceAccessStrategy.java (hunk 4 above)
Comment (DavidRG13) [inline core/cas-server-core-services/src/main/java/org/apereo/cas/services/DefaultRegisteredServiceAccessStrategy.java]: There are, I was fighting against them for a while...

Comment (DavidRG13) [inline core/cas-server-core-services/src/main/java/org/apereo/cas/services/DefaultRegisteredServiceAccessStrategy.java]: If I refactor it, you will lose logs, is it fine?

Comment (DavidRG13) [inline core/cas-server-core-services/src/main/java/org/apereo/cas/services/DefaultRegisteredServiceAccessStrategy.java]: Should we do this in a separate PR? just to be on topic

Comment (mmoayyed) [inline core/cas-server-core-services/src/main/java/org/apereo/cas/services/DefaultRegisteredServiceAccessStrategy.java]: Probably not. This one is closely related, but if you prefer separate PRs that's cool.

Comment (DavidRG13) [inline core/cas-server-core-services/src/main/java/org/apereo/cas/services/DefaultRegisteredServiceAccessStrategy.java]: Done
